### PR TITLE
Collation requires a datetime string

### PR DESCRIPTION
**Pull request by samuelhwilliams**
Also:

Simplify letter collation times

* Remove the ability to specify a print run deadline for the scheduled
celery task.
* Pass the print run deadline through to the celery task in string
format (rather than celery implicitly changing from a datatime to the
string format).
* Convert back from local to UTC when passing through the print run
deadline - this is what the collation task will use to identifying which
notifications to collate.

### CODE REVIEW

File: tests/app/celery/test_letters_pdf_tasks.py [hunk 1]
```diff
@@ -327,24 +327,25 @@ def test_get_key_and_size_of_letters_to_be_sent_to_print_handles_file_not_found(
...
-        ("2021-06-01 12:00:00", datetime(2021, 6, 1, 13, 0, 0)),
-        ("2021-01-01 12:00:00", datetime(2021, 1, 1, 12, 0, 0)),
-        (None, datetime(2021, 6, 1, 17, 30)),
+        # 5:50pm and 5:30pm in GMT, represented in UTC
+        ("2021-01-01 17:50:00+00:00", "2021-01-01T17:30:00"),
+        # 5:50pm and 5:30pm in BST, represented in UTC
+        ("2021-06-01 16:50:00+00:00", "2021-06-01T16:30:00"),
     ],
 )
 def test_check_time_to_collate_letters_respects_print_run_deadline(
```
Comment (leohemsted) [inline tests/app/celery/test_letters_pdf_tasks.py:338]: minor nitpick - the other test of this function - `test_check_time_to_collate_letters_exits_early_outside_of_expected_window` - does a lot of unnecessary setup that looks like a relic from when it was testing a function with wider scope - can we get rid of some of that? (and also ideally colocate the three tests within the file, whether in a class or not i don't mind)

**Review by leohemsted — CHANGES_REQUESTED** [review]

File: app/celery/letters_pdf_tasks.py [hunk 2]
```diff
@@ -131,14 +130,15 @@ def update_validation_failed_for_templated_letter(self, notification_id, page_co
...
-def collate_letter_pdfs_to_be_sent(print_run_deadline):
+def collate_letter_pdfs_to_be_sent(print_run_deadline_utc_str: str):
     """
     Finds all letters which are still waiting to be sent to DVLA for printing
 
     This would usually be run at 5.50pm and collect up letters created between before 5:30pm today
     that have not yet been sent.
     If run after midnight, it will collect up letters created before 5:30pm the day before.
     """
+    print_run_deadline = datetime.fromisoformat(print_run_deadline_utc_str)
```
[the remark this answers is not in the document]
Comment (samuelhwilliams) [reply app/celery/letters_pdf_tasks.py:141]: We need to stop using naive datetimes 🙏

File: tests/app/celery/test_letters_pdf_tasks.py [hunk 3]
```diff
@@ -437,7 +438,7 @@ def test_collate_letter_pdfs_to_be_sent(notify_api, mocker, sample_organisation)
     mock_send_email_to_dvla = mocker.patch("app.celery.letters_pdf_tasks.send_letters_volume_email_to_dvla")
 
     with set_config_values(notify_api, {"MAX_LETTER_PDF_COUNT_PER_ZIP": 2}):
-        collate_letter_pdfs_to_be_sent(datetime(2020, 2, 17, 17, 30, 0))
+        collate_letter_pdfs_to_be_sent("2020-02-17T17:30:00")
```
[the remark this answers is not in the document]
Comment (samuelhwilliams) [reply tests/app/celery/test_letters_pdf_tasks.py:441]: We're freezing time at 6pm but all of the notifications are created with some negative timedelta applied to that.

[the remark this answers is not in the document]
Comment (leohemsted) [reply tests/app/celery/test_letters_pdf_tasks.py]: ah missed that 👍

**Review by leohemsted — APPROVED** [review]

**Pull request merged**